### PR TITLE
Added Python 3.5 “async for” compat to websocket

### PR DESCRIPTION
**Pull request by flying-sheep**
Instead of

```py
while True:
	msg = await ws.receive()
	...
	elif msg.tp == web.MsgType.close:
		break
```

do:

```py
async for msg in ws:
	...
```

### CODE REVIEW

**Comment by flying-sheep** [comment]
uh, those test failures are no my fault. i fixed them in #544

**Comment by flying-sheep** [comment]
triggering rebuild

**Comment by flying-sheep** [comment]
hmm, OK, now the error is:

``` console
$ flake8 aiohttp
aiohttp/web_ws.py:297:19: F821 undefined name 'StopAsyncIteration'
```

but this is wrong. of course this is an undefined name in python < 3.5, but as you’d have to call `ws.__anext__()` manually in those versions, and you don’t do that because it’s a magic method wrapping `receive`.

so i’ll just make flake8 shut up.

**Comment by asvetlov** [comment]
Tests and documentation update are required

**Comment by flying-sheep** [comment]
i don’t seem to get this test to run. it hangs, don’t know why…

**Comment by carlcarl** [comment]
In Travis log: `aiohttp/web_ws.py:295:1: W293 blank line contains whitespace`.
There are spaces at line 295.
(The testing use `flake8` to check the coding style at the beginning)

**Comment by flying-sheep** [comment]
i know, i just made a last-minute fix and didn’t run it after :neutral_face: 

check the new one, that’s the real problem.

**Comment by carlcarl** [comment]
Seems hangs on `test_web_websocket.py` line 42:

``` python
msg = await resp._reader.read()
```

In `resp._reader.read()`, check `streams.py` line 374:

``` python
yield from self._waiter
```

It stops there I guess...

**Comment by flying-sheep** [comment]
ya. why?

**Comment by carlcarl** [comment]
Here is the part of the my testing in `test_await`:

``` python
    items = ['q1', 'q2', 'q3']
    for item in items:
        resp._writer.send(item)
        msg = await resp._reader.read()
        assert msg.tp == websocket.MSG_TEXT
        assert item + '/answer' == msg.data

    resp._writer.close()

    msg = await resp._reader.read()
    assert msg.tp == websocket.MSG_CLOSE
    assert msg.data == 1000
    assert msg.extra == ''

    await closed
    await resp.close()
```

Server will not know if it need to send close message back. So you need to send close first or it will wait for `receive` forever.

Also, remember to `await resp.close()`

**Comment by flying-sheep** [comment]
thanks! you saved me quite some time. :bow: 

i copied the test from elsewhere and adapted it … poorly.

i hope this is it then!

File: tests/test_py35/test_web_websocket_35.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        closed.set_result(1)
+        return ws
+
+    port = unused_port()
+    await create_server(loop, port, 'GET', '/', handler)  # returns server
+    resp = await ws_connect('ws://127.0.0.1:{p}'.format(p=port), loop=loop)
+
+    items = ['q1', 'q2', 'q3']
+    for item in items:
+        resp._writer.send(item)
```
Comment (asvetlov) [inline tests/test_py35/test_web_websocket_35.py:37]: Why do you need private attribute access. The test may use `send()` and `receive()` client response methods.

Comment (flying-sheep) [inline tests/test_py35/test_web_websocket_35.py:37]: eh, no idea, i copied stuff from elsewhere. what methods are you referring to?

Comment (carlcarl) [inline tests/test_py35/test_web_websocket_35.py:37]: `resp._writer.send(item)` ?
I think he means `_writer` is a private attribute and should not be used in testing.

Comment (flying-sheep) [inline tests/test_py35/test_web_websocket_35.py:37]: sure. my question is what i should use instead

Comment (asvetlov) [inline tests/test_py35/test_web_websocket_35.py:37]: http://aiohttp.readthedocs.org/en/stable/client_websockets.html

**Comment by flying-sheep** [comment]
oh, sorry. i would have done it eventually :sweat_smile:

**Pull request merged**